spec/plaindatetime.html

```diff
@@ -28,19 +28,18 @@ <h1>Temporal.PlainDateTime ( _isoYear_, _isoMonth_, _isoDay_ [ , _hour_ [ , _min
       <p>
         This function performs the following steps when called:
       </p>
-      <emu-note>The value of ! ToIntegerWithTruncation(*undefined*) is 0.</emu-note>
       <emu-alg>
         1. If NewTarget is *undefined*, then
           1. Throw a *TypeError* exception.
         1. Let _isoYear_ be ? ToIntegerWithTruncation(_isoYear_).
         1. Let _isoMonth_ be ? ToIntegerWithTruncation(_isoMonth_).
         1. Let _isoDay_ be ? ToIntegerWithTruncation(_isoDay_).
-        1. Let _hour_ be ? ToIntegerWithTruncation(_hour_).
-        1. Let _minute_ be ? ToIntegerWithTruncation(_minute_).
-        1. Let _second_ be ? ToIntegerWithTruncation(_second_).
-        1. Let _millisecond_ be ? ToIntegerWithTruncation(_millisecond_).
-        1. Let _microsecond_ be ? ToIntegerWithTruncation(_microsecond_).
-        1. Let _nanosecond_ be ? ToIntegerWithTruncation(_nanosecond_).
+        1. If _hour_ is *undefined*, let _hour_ be *+0*<sub>𝔽</sub>; else let _hour_ be ? ToIntegerWithTruncation(_hour_).
+        1. If _minute_ is *undefined*, let _minute_ be *+0*<sub>𝔽</sub>; else let _minute_ be ? ToIntegerWithTruncation(_minute_).
+        1. If _second_ is *undefined*, let _second_ be *+0*<sub>𝔽</sub>; else let _second_ be ? ToIntegerWithTruncation(_second_).
+        1. If _millisecond_ is *undefined*, let _millisecond_ be *+0*<sub>𝔽</sub>; else let _millisecond_ be ? ToIntegerWithTruncation(_millisecond_).
+        1. If _microsecond_ is *undefined*, let _microsecond_ be *+0*<sub>𝔽</sub>; else let _microsecond_ be ? ToIntegerWithTruncation(_microsecond_).
+        1. If _nanosecond_ is *undefined*, let _nanosecond_ be *+0*<sub>𝔽</sub>; else let _nanosecond_ be ? ToIntegerWithTruncation(_nanosecond_).
```

Comment:
This was a pre-existing issue, but as long as you are editing this algorithm we may as well fix it: these should read "set _hour_ to 0" and "set _hour_ to ? ToIntegerWithTruncation(_hour_)", etc., since the variables already existed.

I think it has to be 0 in this case, not +0<sub>𝔽</sub> though. There are two approaches possible:

- If _hour_ is **undefined**, set _hour_ to **+0**<sub>𝔽</sub>.
- Set _hour_ to ? ToIntegerWithTruncation(_hour_).

or, as you have it now,

- If _hour_ is **undefined**, set _hour_ to 0; else set _hour_ to ? ToIntegerWithTruncation(_hour_).

Either one is fine, but in each case _hour_ ends up being 0 in the ℝ domain at the end, but in the first approach it has to start in the 𝔽 domain because ToIntegerWithTruncation takes 𝔽 and returns ℝ.